devtools/common/src/main/java/io/quarkus/cli/commands/file/GradleBuildFile.java

```diff
@@ -130,20 +115,20 @@ private void completeSettingsContent(String artifactId) {
             res.append(System.lineSeparator()).append("rootProject.name='").append(artifactId).append("'")
                     .append(System.lineSeparator());
         }
-        settingsContent = res.toString();
+        getModel().setSettingsContent(res.toString());
     }
 
-    private void completeProperties() {
-        if (propertiesContent.getProperty("quarkusVersion") == null) {
-            propertiesContent.setProperty("quarkusVersion", getPluginVersion());
+    private void completeProperties() throws IOException {
+        if (getModel().getPropertiesContent().getProperty("quarkusVersion") == null) {
+            getModel().getPropertiesContent().setProperty("quarkusVersion", getPluginVersion());
         }
     }
 
     @Override
-    protected void addDependencyInBuildFile(Dependency dependency) {
+    protected void addDependencyInBuildFile(Dependency dependency) throws IOException {
         StringBuilder newBuildContent = new StringBuilder();
-        readLineByLine(buildContent, new AppendDependency(newBuildContent, dependency));
-        buildContent = newBuildContent.toString();
+        readLineByLine(getModel().getBuildContent(), new AppendDependency(newBuildContent, dependency));
```

Comment:
while it's not an issue with this PR specifically, I was looking into the code for AppendDependency and I noticed 2 corner cases that don't currently work:
1. It looks for `line.startsWith("dependencies {"` which doesn't account for additional spacing between `dependencies` and `{` and also doesn't account for leading whitespace (which is normal if this is being applied in a `subprojects{}` block for example).
1. If we have the quarkus plugin installed already, but no `dependencies{}` at all, nothing gets added. For this case I think we should probably add the quarkus-bom dependency with a version corresponding to the quarkus-gradle-plugin, followed by the rest of the requested extension dependencies as normal

Reply:
TBH I didn't really look at the business or the code itself, and I don't really want to.. I just fixed the bug using the same system used in MavenBuildFile and added the tests that should have been there in the first place..

Reply:
At some point we are going to rewrite most of the codegen part with a real template engine for Quarkus that is beeing developed https://github.com/quarkusio/qute.

But for now, any corner case you see, please feel free to provide a PR with some tests :)

Reply:
Yeah this part of the code is messy currently... So trying to improve little things probably doesn't make a ton sense, tests on the other hand definitely do :).
@ia3andy did you sign up for rewriting the part this `qute`? :P

Reply:
@geoand why not, that might be fun, though for now I have plenty on my plate 😅